lib/dynamodb-integration.js

```diff
@@ -0,0 +1,83 @@
+var LIMITS = {
+    MAX_EVENT_MSG_SIZE_BYTES: 400000,   // The real max size is 409,600, we leave some room for overhead on each message
+    MAX_BATCH_SIZE_BYTES: 16000000,     // We leave some fudge factor here too. This shouldn't be reachable for 25 items, however.
+    MAX_BATCH_ITEM_NUM: 25
+}
+
+// CloudWatch adds 26. DynamoDB doesn't, but we wanna add a constant size to each item
+// This size should depend on the attribute length and non-message values saved in DB
+var BASE_EVENT_SIZE_BYTES = 26;
+
+var debug = require('./utils').debug;
+
+var lib = {};
+
+lib.upload = function(aws, tableName, streamName, logEvents, options, cb) {
+    debug('upload', logEvents);
+
+    var entryIndex = 0;
+    var bytes = 0;
+    while (entryIndex < Math.min(logEvents.length, LIMITS.MAX_BATCH_ITEM_NUM)) {
+        var ev = logEvents[entryIndex];
+        // unit tests pass null elements
+        var evSize = ev ? Buffer.byteLength(ev.message, 'utf8') + BASE_EVENT_SIZE_BYTES : 0;
+        if(evSize > LIMITS.MAX_EVENT_MSG_SIZE_BYTES) {
+            evSize = LIMITS.MAX_EVENT_MSG_SIZE_BYTES;
+            ev.message = ev.message.substring(0, evSize);
+            const msgTooBigErr = new Error('Message Truncated because it exceeds the DynamoDB size limit');
+            msgTooBigErr.logEvent = ev;
+            cb(msgTooBigErr);
+        }
+        if (bytes + evSize > LIMITS.MAX_BATCH_SIZE_BYTES) break;
+        bytes += evSize;
+        entryIndex++;
+    }
+
+    const eventsToSend = logEvents.splice(0, entryIndex);
+    const payload = {
+        RequestItems: {
+            [tableName]: eventsToSend.map(event =>
+                ({
```

Comment:
In this PR, yes. I'll change the `timestamp` to be a `sequenceNumber` in a next PR, because of [this issue](https://github.com/env0/winston-aws-dynamodb/issues/3)